### ISSUE TITLE
handling chunks

### ISSUE THREAD

**Issue by AlvinTam-lab** [nanoporetech/bonito#424]
Hi Bonito team,
I’m using the bonito train command and had a few questions:
- Is there a way to automatically use all the chunks within chunks.npy without needing to manually specify the total number of chunks?
- If I want to split the chunks into a training set and a validation set, is there a built-in functionality to handle the splitting? Or do I need to manually divide them before passing them to Bonito?
- Let's say my chunks.npy contains 202,000 chunks, and I want to reserve 2,000 chunks for validation. In this case, should I set --chunks to 202000 or 200000?

Thanks!

**Comment by davidnewman02** [comment]
- Bonito will load all the chunks in the dataset and randomly sample them throughout training, where the length of training is controlled by the number `--epochs` multiplied by the number of `--chunks`. There's not a direct method to "pass through all chunks N times in order/without-replacement",
- It is possible to make a separate set of validation chunks. If there is a subfolder in the `--directory` named `validation` then these chunks will be used as the validation chunks otherwise the last 3% of the dataset is used for validation. (see [data.py#L104](https://github.com/nanoporetech/bonito/blob/master/bonito/data.py#L104) )
- If you have 200k chunks then you could set `--epochs 1 --chunks 200_000` and you would expect to see each chunk on average once. You can separate your validation chunks as described above.

The relevant code sections are:
- load_data: https://github.com/nanoporetech/bonito/blob/master/bonito/data.py#L32
  - checks for the presence of `chunks.npy` to load
- load_numpy: https://github.com/nanoporetech/bonito/blob/master/bonito/data.py#L99
  - loads the `chunks.npy` as a memmapped numpy file 
  - creates the DataLoader args and shuffling
- load_data: https://github.com/nanoporetech/bonito/blob/master/bonito/data.py#L67
  - creates the DataLoader using the args specified 


Alternatively if you require direct control of the chunks selection/ordering you could consider providing your own DataLoader as described here: https://github.com/nanoporetech/bonito/blob/master/documentation/training.md#providing-data-dynamically-with-datasetpy
